### ISSUE TITLE
Nuitka should test for requirements (like dependency tree walker) before starting build

### ISSUE THREAD

**Issue by jimkring** [Nuitka/Nuitka#3060]
I am trying to get a new CI build process going and I forgot to include `--assume-yes-for-downloads` in my `nuikta` build step.  Nuikta took many minutes (for compilation to complete) before failing the build (at the linking step) stating that the dependency walker was required and would need to be downloaded.

Additionally, I don't think the error message suggested adding the `--assume-yes-for-downloads` option. I only got an error at the step where it asks if it's OK to download.

So, this is two very related requests, please:

1) Check if downloadable requirements are satisfied (and do the prompting and exception raising) prior to starting the build, so that users know ASAP that their build will fail.

2) Add something to the error message about the `--assume-yes-for-downloads` when doing a CI/unattended build, since it took me a while to search for this option on the website docs.

**Comment by kayhayen** [comment]
I agree with these suggestions. Nuitka-Action I think does this, but not everybody uses it, or should even have to use it. Adding these things should be easy, but I guess it's hard to be come complete about it.
